### PR TITLE
Electrostatic Potential

### PR DESCRIPTION
**Pull request by iglesias-cardinale**
Added a module to find electrostatic potential from a set charge distribution. This could be expanded to include continuous charge distributions. Module includes lambda functions and iterators (np.enumerate). 

Added brief documentation on the theory of electrostatic potential. This should be added to.

Added unit test module to test
- Vector potential is zero when charge is zero
- Potential due to single charge at origin matched Coulomb's law
- Potential function returns infinity if charge and observation point overlap.

### CODE REVIEW

File: __pycache__/electrostatics.cpython-312.pyc
Comment (HeshamElsaman) [inline __pycache__/electrostatics.cpython-312.pyc:1]: Please, remove this file.

**Review by HeshamElsaman — CHANGES_REQUESTED** [review]
- I'm actually not sure about linting, even for myself, but the code is well-functioning.
- Two of the functional programming concepts were utilized.
- Modules are well-separated.
However,
- The files need to be moved into a different folder not to possibly conflict with similarly named files from other people and for a better modularity.
- A binary file needs to be removed. It probably happened because the PR was created before updating the branch to include the .gitignore file.

**Comment by laserlab** [comment]
> * I'm actually not sure about linting, even for myself, but the code is well-functioning.

The main linting complain that you could have picked up too for the review are the missing docstrings.

File: electrostatics.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import numpy as np
+
+#Constants
+epsilon0 = 8.854187817e-12  #Farads per meter
+
+#Lambda function to calculate the distance between an observation point x and all charges p
+r_xp = lambda x, p: np.linalg.norm(x - p, axis=1)
```
Comment (laserlab) [inline electrostatics.py:7]: Agree with the linter here. Not useful use of lambda function. Also it doesn’t fulfill the homework requirement of using your ED function as an argument in another function

**Review by laserlab — CHANGES_REQUESTED** [review]
- linting
- lambda function use
- no ED function as argument in another function

**Pull request closed without merging**